static/js/publisher/release/selectors/index.js

```diff
@@ -282,19 +287,53 @@ export function getProgressiveState(state, channel, arch, revision) {
     item => revision === item.revision
   );
 
-  let progressive = null;
-
   const release = allReleases[releaseIndex];
 
-  if (release && release.progressive && release.progressive.key) {
-    progressive = release.progressive;
+  if (
+    release &&
+    release.progressive &&
+    release.progressive &&
+    release.progressive.key
+  ) {
+    progressiveStatus = JSON.parse(JSON.stringify(release.progressive));
+
+    previousRevision = allReleases
+      .slice(releaseIndex)
+      .filter(item => item.revision !== release.revision);
+
+    if (previousRevision[0]) {
+      previousRevision = revisions[previousRevision[0].revision];
+    }
+  }
+
+  const pendingMatch = pendingReleases[revision]
+    ? pendingReleases[revision][channel]
+    : undefined;
 
-    // TODO: Confirm that this is true when multiple progressive
-    // releases have occured.
-    if (allReleases[releaseIndex + 1]) {
-      progressive.from = allReleases[releaseIndex + 1].revision;
+  if (pendingMatch) {
+    if (
+      channel === pendingMatch.channel &&
+      pendingMatch.revision.architectures.includes(arch)
+    ) {
+      pendingProgressiveStatus = Object.assign({}, pendingMatch.progressive);
     }
   }
 
-  return progressive;
+  return [progressiveStatus, previousRevision, pendingProgressiveStatus];
```

Comment:
The comment for this function could be extended to explain better what is returned, especially that it's not just a single object but 3 values now.